### PR TITLE
GA ADL Update

### PR DESCRIPTION
**Pull request by begoldsm**
This update brings our Swagger specs in line with the new strict Swagger guidance, which also has the added benefit of making the code more clean and usable. Below are the big things that have changed:
For all the swagger spec changes and feedback please see this PR: https://github.com/Azure/azure-rest-api-specs/pull/635
1. All nested properties have been flattened that can be
2. Operations have been branched out into separate operation classes related to the resources they impact
3. All enums have been properly cased.
4. All objects now properly have required properties, defaults, min and max values.
5. PATCH operations now properly only allow patching of the specific things that can be patched.
6. There is documentation for all responses now.
7. All tests were re-run and are passing, and I added some new negative tests.

### CODE REVIEW

**Comment by begoldsm** [comment]
@markcowl can we get someone to look at this? I know it is quite large, but the majority of the files are test/recording files. I would like to get this in today, if possible, so I can work next week on getting everything needed fixed up in PowerShell for the december release.

**Comment by markcowl** [comment]
@begoldsm someone will look at it today or on Monday.  BTW, can you include a link to the corresponding spec changes?

**Comment by begoldsm** [comment]
@markcowl here is the link to the corresponding swagger:
https://github.com/Azure/azure-rest-api-specs/pull/635

**Comment by begoldsm** [comment]
@markcowl can someone please look at this and get it merged or commented on?

File: src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/Models/AddDataLakeStoreParameters.cs [hunk 1]
```diff
@@ -29,32 +30,18 @@ public partial class AddDataLakeStoreParameters
...
         /// Initializes a new instance of the AddDataLakeStoreParameters class.
         /// </summary>
-        /// <param name="properties">the properties for the Data Lake Store
-        /// account being added.</param>
-        public AddDataLakeStoreParameters(DataLakeStoreAccountInfoProperties properties)
+        /// <param name="suffix">the optional suffix for the Data Lake Store
+        /// account.</param>
+        public AddDataLakeStoreParameters(string suffix = default(string))
         {
-            Properties = properties;
```
Comment (cormacpayne) [inline src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/Models/AddDataLakeStoreParameters.cs]: @begoldsm this comment holds for all classes in Model

File: src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/project.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "0.12.6-preview",
+  "version": "1.0.0",
```
Comment (cormacpayne) [inline src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/project.json]: @begoldsm this comment holds for DataLake.Store as well

**Review by cormacpayne — CHANGES_REQUESTED** [review]
@begoldsm more comments here since I am unable to open all of the files on GitHub:

The below comments apply to both DataLake.Analytics and DataLake.Store
- Is there sufficient testing for the changes made to the generated code? It appears there won't too many changes made in ScenarioTests
- There are many classes missing in the Model folder; you will need to mitigate this in PowerShell if any of them are a part of a parameter or part of any output type
- The generate.cmd file has not been updated with the AutoRest version you used
- The generate.cmd file should link to the ADL swagger specs in the Azure fork, not yours

File: src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/project.json (hunk 2 above)
Comment (begoldsm) [inline src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/project.json]: @cormacpayne I need more information from you on the release out of preview. I met all the requirements in the swagger spec review (see the linked spec review). That is also why there are so many "breaking" changes and why this is being updated to 1.0.0. This is a HUGE change from our previous preview packages, which is why we wanted to get this in ASAP. As you mentioned, I will need to mitigate all the SDK breaking changes in PowerShell, so I need the necessary time to do so. I will address the update comments and regenerate, but for all breaking changes I will address that in PowerShell until the next round of allowed breaking changes when I will update everything to be flattened in PowerShell.

**Review by cormacpayne — APPROVED** [review]

**Comment by cormacpayne** [comment]
:shipit:

**Pull request merged**